main.nf

```diff
@@ -180,29 +200,113 @@ process get_software_versions {
 }
 
 /*
- * STEP 0: Extract reads mapping to specific chromosome(s)
+ * If index_files not provided as input
  */
-if (params.chr){
-  process extractReadsMappingToChromosome{
-    tag "${name}.${chr_list_joined}"
-    label 'process_medium'
+
+if(!params.index_files){
+	process IdxBAI {	
+		tag "$name"
+		label 'process_medium' 
+
+		input:
+		set val(name), file(bam) from bam_files_index		
+	
+		output:
+		file "${name}.bai" into ch_index_files 
+		file "${name}.bai" into ch_chr_index_files
+	
+		
+		script:	
+		"""		
+		samtools index $bam > ${name}.bai	
+		"""		
+	}
+	
+	// Extract reads mapping to specific chromosome(s)
+	if (params.chr){
+	  process extractReadsMappingToChromosome{
+	    tag "${name}.${chr_list_joined}"
+	    label 'process_medium'
   
-    input:
-    set val(name), file(bam) from bam_chr 
+	    input:
+	    set val(name), file(bam) from bam_chr 
+		  set val(name), file(bai) from ch_chr_index_files
+
+	    output:
+	    set val("${name}.${chr_list_joined}"), file("${name}.${chr_list_joined}.bam") into bam_files_check
+		                
+
+	    script:
+	    //If multiple chr were specified, then join space separated list for naming: chr1 chr2 -> chr1_chr2, also resolve region specification with format chr:start-end
+	    chr_list_joined = params.chr.split(' |-|:').size() > 1 ? params.chr.split(' |-|:').join('_') : params.chr
+	    """    
+	    samtools view -hb $bam ${params.chr} -@$task.cpus -o ${name}.${chr_list_joined}.bam
```

Comment:
```suggestion
	    samtools view -hb $bam ${params.chr} -@$task.cpus -o "${name}.${chr_list_joined}.bam"
```

Reply:
I've seen before that not adding the `"` when having a string that contains variables inside but also additional characters can give rise to problems.